File: gcd.js
function gcd(p, q) {
    p
    q
    if (q == 0) {
        return p
    } else {
        return gcd(q, p % q)
    }
}

const gcd = (p, q) => q === 0 ? p : gcd (q, p % q)
// z = gcd(105, 24)
// z

z = gcd(999,33)
z

File: 7kyu/Greatest common divisor/GCD.java
/*
Find the greatest common divisor of two positive integers.
The integers can be large, so you need to find a clever solution.

The inputs x and y are always greater or equal to 1, so the the greatest common divisor
will always be an integer that is also greater or equal to 1.
*/
/*

public class GCD {
    public static int compute(int x, int y) {
        if (y == 0) return x;
        return compute(y, x % y);
    }
}*/

public class GCD {
    public static int compute(int x, int y) {
        return y == 0 ? x : compute(y, x % y);
    }
}
